src/main/java/htsjdk/samtools/SAMSequenceRecord.java

```diff
@@ -184,6 +220,7 @@ public boolean equals(final Object o) {
         if (mSequenceLength != that.mSequenceLength) return false;
         if (!attributesEqual(that)) return false;
         if (mSequenceName != that.mSequenceName) return false; // Compare using == since we intern() the name
+        // TODO: should this also compare the alternative names?
```

Comment:
it seems strange to leave alternative names out of equals especially given the weaker `isSameSequence()` also exists